composer/trainer/dist_strategy.py

```diff
@@ -319,31 +304,26 @@ def sync_hook(*args):
     sharding_strategy = SHARDING_MAP[sharding_map_key]
 
     kwargs = {}
-    if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0'):
-        if 'device_mesh' in fsdp_config:
-            device_mesh_size = len(fsdp_config['device_mesh'])
-            if sharding_strategy in [
-                ShardingStrategy.FULL_SHARD,
-                ShardingStrategy.SHARD_GRAD_OP,
-                ShardingStrategy.NO_SHARD,
-            ] and device_mesh_size != 1:
-                raise ValueError(
-                    f'FSDP sharding strategy {sharding_map_key.upper()} requires a device mesh '
-                    f'of size 1 but got device mesh size of {device_mesh_size}.',
-                )
-            elif sharding_strategy in [
-                ShardingStrategy.HYBRID_SHARD,
-                ShardingStrategy._HYBRID_SHARD_ZERO2,
-            ] and device_mesh_size != 2:
-                raise ValueError(
-                    f'FSDP sharding strategy {sharding_map_key.upper()} requires a device mesh '
-                    f'of size 2 but got device mesh size of {device_mesh_size}.',
-                )
-            from torch.distributed._tensor import init_device_mesh
-            kwargs['device_mesh'] = init_device_mesh(
-                'cuda',
-                tuple([int(x) for x in fsdp_config['device_mesh']]),
+    if version.parse(torch.__version__.split('.dev')[0]) >= version.parse('2.2.0') and 'device_mesh' in fsdp_config:
+        if 'process_group' not in fsdp_config:
+            warnings.warn(
+                'process_group and device_mesh are set for FSDP, so ignoring device_mesh. Please set process_group to None.',
```

Comment:
If we're recommending setting process_group to None, and also defaulting to using `DeviceMesh` everywhere, why do we ignore `device_mesh` here?

Reply:
We autoset device mesh, so if you specify FSDP, it would have both and thus the user specified one is likely process_group... 🤔. It's pretty hacky...

Reply:
I thought we were going to be completely moving to `device_mesh` so that we move to DTensor fully. Does using `process_group` go against that

Reply:
Yes, but we have to deprecate. we cant just delete :(